File: Search_Insert_Position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int l = 0;
        int r = nums.size()-1;
        int mid;
        if(nums[r] < target) return r+1;
        while(l+1 < r)
        {
            mid = (r+l)/2;
            if(nums[mid] == target) return mid;
            if(nums[mid] > target)
            {
                r = mid;
            }
            else
            {
                l = mid;
            }
        }
        return nums[l] < target ? l+1 : l;
    }
};

File: Search_Insert_Position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        left = 0
        right = len(nums) - 1
        while right - left > 1:
            mid = (left + right) // 2
            if nums[mid] == target:
                return mid
            if nums[mid] < target:
                left = mid
            else:
                right = mid
        if nums[right] < target:
            return right + 1
        if nums[left] >= target:
            return left
        return right
